[ICDS Dashboard]: Back filling LS Launched in the PS and map views

ShubhamBansal1997 opened the pull request:
<!--- Provide a link to the ticket or document which prompted this change -->

##### SUMMARY
https://app.asana.com/0/1179234577945697/1178662132203953/f
<!--- Describe the change below, including rationale and design decisions -->

##### FEATURE FLAG
<!--- If this is specific to a feature flag, which one? -->

##### RISK ASSESSMENT / QA PLAN
<!-- Does this need QA before or after merge? Link QA ticket. -->

##### PRODUCT DESCRIPTION
<!--- For non-invisible changes, describe user-facing effects. -->

ShubhamBansal1997 committed:
🎉 added the script for the past data correction ls

Rohit25negi commented on custom/icds_reports/management/commands/fix_ls_data.py:
```diff
@@ -0,0 +1,87 @@
+                FROM "{from_table}"
+                GROUP BY {group_by}, month
+            ) ut
+            WHERE ls.{update_location} = ut.{update_location}
```
good to compare agg_level and month as well here. Or you think that will automatically be taken care by the table names?

ShubhamBansal1997 replied:
set of queries gonna run for a month
```
CREATE UNLOGGED TABLE "tmp_ls_usage" AS SELECT
    supervisor_id,
    count(*) as form_count
    FROM "ucr_icds-cas_static-ls_usage_forms_85eca43d" ls_usage_ucr
    WHERE timeend<'2017-04-01'
    GROUP BY supervisor_id


UPDATE "agg_ls_2017-04-01_4" agg_ls
    SET num_supervisor_launched = CASE WHEN form_count>0 THEN 1 ELSE 0 END
    from (
        SELECT
            supervisor_id,
            form_count
        FROM tmp_ls_usage
        ) ut
        WHERE agg_ls.supervisor_id = ut.supervisor_id


DROP TABLE "tmp_ls_usage"



            UPDATE "agg_ls_2017-04-01_3" ls
            SET
            ls.num_supervisor_launched = ut.num_supervisor_launched
            FROM (
                SELECT
                block_id,
                sum(num_supervisor_launched) as num_supervisor_launched
                FROM "agg_ls_2017-04-01_4"
                GROUP BY state_id,district_id,block_id, month
            ) ut
            WHERE ls.block_id = ut.block_id



            UPDATE "agg_ls_2017-04-01_2" ls
            SET
            ls.num_supervisor_launched = ut.num_supervisor_launched
            FROM (
                SELECT
                district_id,
                sum(num_supervisor_launched) as num_supervisor_launched
                FROM "agg_ls_2017-04-01_3"
                GROUP BY state_id,district_id, month
            ) ut
            WHERE ls.district_id = ut.district_id



            UPDATE "agg_ls_2017-04-01_1" ls
            SET
            ls.num_supervisor_launched = ut.num_supervisor_launched
            FROM (
                SELECT
                state_id,
                sum(num_supervisor_launched) as num_supervisor_launched
                FROM "agg_ls_2017-04-01_2"
                GROUP BY state_id, month
            ) ut
            WHERE ls.state_id = ut.state_id
```

Rohit25negi reviewed (approved):
overall looks good. Left two questions

ShubhamBansal1997 commented on custom/icds_reports/management/commands/fix_ls_data.py:
```diff
@@ -0,0 +1,87 @@
+    def rollup_query(self, agg_level, month):
+        locations = ['state_id', 'district_id', 'block_id', 'supervisor_id']
+
+        for i in range(3, agg_level - 1, -1):
```
using the array below as a param

ShubhamBansal1997 committed:
✏️ minor typo fix in the query

ShubhamBansal1997 committed:
🐛 minor fixes

- putting month increments at correct place
- correcting the spliting the string
- correcting the loop date
